lib/auth/auth_with_roles.go

```diff
@@ -1497,6 +1548,14 @@ func (a *ServerWithRoles) ListUnifiedResources(ctx context.Context, req *proto.L
 		}
 	}
 
+	if req.UseSearchAsRoles && len(resourceAccess.requestableResourceNames) > 0 {
+		for _, r := range paginatedResources {
+			if slices.Contains(resourceAccess.requestableResourceNames, r.GetResourceName()) {
+				r.RequiresRequest = true
+			}
+		}
+	}
```

Comment:
I'll switch to a set or map here.

I was having trouble doing the paginating inline during the iteration from the unified cache but I think I can create the item inline and sorta ignore the returned resources from the Get. I'll mess around a bit more but definitely will at least do the set map thing. Thanks!

Reply:
Ok, second pass at it here https://github.com/gravitational/teleport/pull/40328/commits/21901901c68b73308f813be962c73a2024e38b5b

I'm making the "match" inside and appending a slice that's used as the results rather than having to do the whole `requestableResourceNames` and `ResourceName()` song and dance.

I'm curious now if I can include the `IncludeLogins` part of it here too but I haven't dove into that yet while I await feedback first